sympy/ntheory/elliptic_curve.py

```diff
@@ -263,6 +279,26 @@ def rank(self):
             return self._rank
         raise NotImplementedError("Still not implemented")
 
+    def div_poly(self, n):
+        """
+        Division Polynomials of an elliptic curve over finite fields.
+
+        """
+        from sympy import simplify
+        if self._domain == QQ:
+            raise ValueError("Only defined for curves over finite fields")
+        if n in self.div:
+            return self.div[n]
+        f = lambda a : self.div_poly(a)
+        if n % 2 == 0:
+            n //= 2
+            self.div[2*n] = f(n) * (f(n + 2)*f(n - 1)**2 - f(n - 2)*f(n + 1)**2)
+            self.div[2*n] = simplify((self.div[2*n] * mod_inverse(2, self.modulus)) / self.Y)
```

Comment:
```suggestion
            self.div[2*n] = (self.div[2*n] * mod_inverse(2, self.modulus)) / self.Y).simplify()
```
Not sure if this will work but `simplify` is slow, so `.simplify()` should be used.